core/arch/x86_code.c

```diff
@@ -442,263 +442,3 @@ safe_read_resume_pc(void)
 {
     return (app_pc)&safe_read_asm_recover;
 }
-
```

Comment:
There is an option I need to add to compile this code that is different for gcc and clang (clang frontend does not support yet, unusual but happens). I can't add the option to the target, because then the clang build fails, because of i#3458 (g++ is used, even in clang build). If I add the option to the x86_code module, then the option is added even to libdynamorio, which I don't want either. So the best way is to put the tests in a separate module that's only part of the unit_test.

So it's actually for the two reasons above that I moved the tests, not only i#3458. I'll rephrase the comment.

Reply:
> If I add the option to the x86_code module, then the option is added even to libdynamorio

unit_tests is a separate build so the option would not be added to libdynamorio.

Reply:
Oh you mean as a source file property in cmake -- ok, I guess there is no way to do source-for-target-only.  If this happens again though I would think it would be better to fix #3458 than to move more code around.

Reply:
> than to move more code around.

(Because we can't test static functions of a module in a separate file, having separate _test files is not viable)